pkg/apis/monitoring/v1/alertmanager_types.go

```diff
@@ -38,7 +38,11 @@ const (
 // +kubebuilder:printcolumn:name="Paused",type="boolean",JSONPath=".status.paused",description="Whether the resource reconciliation is paused or not",priority=1
 // +kubebuilder:subresource:status
 
-// Alertmanager describes an Alertmanager cluster.
+// The `Alertmanager` custom resource definition (CRD) declaratively defines a desired [Alertmanager](https://prometheus.io/docs/alerting) setup to run in a Kubernetes cluster. It provides options to configure the number of replicas and persistent storage.
+//
+// For each `Alertmanager` resource, the Operator deploys a `StatefulSet` in the same namespace. The Alertmanager pods are configured to mount a `Secret` called `alertmanager-<alertmanager-name>` which holds the Alertmanager configuration under the key `alertmanager.yaml`.
+//
+// When there are two or more configured replicas, the Operator runs the Alertmanager instances in high-availability mode.
```

Comment:
(suggestion) the secret part is a bit outdated since you can also use AlertmanagerConfig objects.

```suggestion
// For each `Alertmanager` resource, the Operator deploys a `StatefulSet` object in the same namespace. When there are two (or more) replicas, it configures automatically the Alertmanager pods in high-availability mode.
```